app/models/concerns/user_invitation.rb

```diff
@@ -0,0 +1,124 @@
+require 'active_support/concern'
+
+module UserInvitation
+  extend ActiveSupport::Concern
+
+  included do
+  	attr_accessor :invitation_role, :invitation_text
+  	devise :invitable
+  	after_invitation_created :create_team_user_invitation
+
+	  def self.send_user_invitation(members, text=nil)
+	    msg = {}
+	    members.each do |role, emails|
+	      emails.split(',').each do |email|
+	        email.strip!
+	        u = User.where(email: email).last
+	        begin
+	          if u.nil?
+	            user = User.invite!({:email => email, :name => email.split("@").first, :invitation_role => role, :invitation_text => text}, User.current) do |iu|
+	              iu.skip_invitation = true
+	            end
+	            user.update_column(:raw_invitation_token, user.raw_invitation_token)
+	            msg[email] = 'success'
+	          else
+	            u.invitation_role = role
+	            u.invitation_text = text
+	            msg.merge!(u.invite_existing_user)
+	          end
+	        rescue StandardError => e
+	          msg[email] = e.message
+	        end
+	      end
+	    end
+	    msg
+	  end
+
+	  def invite_existing_user
+	    msg = {}
+	    unless Team.current.nil?
+	      tu = TeamUser.where(team_id: Team.current.id, user_id: self.id).last
+	      if tu.nil?
+	        options = {}
+	        unless self.is_invited?
+	          raw, enc = Devise.token_generator.generate(User, :invitation_token)
+	          options = {:enc => enc, :raw => raw}
+	        end
+	        create_team_user_invitation(options)
+	        msg[self.email] = 'success'
+	      elsif tu.status == 'invited'
+	        msg[self.email] = 'This email already invited to this team'
+	      else
+	        msg[self.email] = 'This email already a team member'
+	      end
+	    end
+	    msg
+	  end
+
+	  def self.accept_team_invitation(token, slug, options={})
+	    # TODO: localize and review error messages copy.
```

Comment:
plz let me know if I've missed some copy

Reply:
All is OK now and I localized all strings